### PR TITLE
Added delete endpoint for company

### PR DESCRIPTION
**Pull request by webbyfox**
To do (delete all that do not apply):

 - [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.

### CODE REVIEW

File: directory_api_client/company.py [hunk 1]
```diff
@@ -188,3 +189,9 @@ def collaboration_request_delete(self, sso_session_id, request_key):
             url=url_collaboration_request_detail.format(request_key=request_key),
             authenticator=self.authenticator(sso_session_id),
         )
+
+    def delete_company_by_sso_id(self, sso_session_id, request_key):
```
Comment (depsiatwal) [inline directory_api_client/company.py:193]: is is this sso_session_id or sso_id ?

**Review by depsiatwal — COMMENTED** [review]
LGTM just small comment

File: directory_api_client/company.py (hunk 1 above)
Comment (webbyfox) [reply directory_api_client/company.py:193]: ahh `sso_id`. give me a min

**Review by depsiatwal — APPROVED** [review]
lgtm

**Pull request merged**